### PR TITLE
local dev/hacking improvements

### PR DESCRIPTION
**Pull request by cgwalters**
This fixes the `./hack/cluster.push` script to replace all MCO components rather than accepting a `WHAT` since...a lot of nontrivial changes require doing all of them, and since we have a uni-image we might as well now!

The `build-image` script changes apply to both the "cluster push fast-path" as well as the "new release image" path.

The `HACKING.md` doc is updated for the above and more.

### CODE REVIEW

**Comment by cgwalters** [comment]
Side note: One thing I struggled with for a bit is that our existing docs encourage pushing to `quay.io` which you need a secret to do...but having that in your `~/.docker/config.json` conflicts with using the OpenShift pull secret to get released containers from `quay.io`.  

Probably for most people with this workflow, it all works fine because your are probably keeping your `~/.docker/config.json` separate from the pull secrets used for openshift-installer.  But [my installer wrapper](https://github.com/cgwalters/xokdinst) links these two by [automatically injecting the docker secrets](https://github.com/cgwalters/xokdinst/commit/8c2308d4bf0b02cd38f20323d551d4c5bcc0b40f#diff-25d902c24283ab8cfbac54dfa101ad31) which creates a conflict.

For OpenShift developers we could work around this probably by encouraging pushing dev images to `registry.svc.ci.openshift.org`?  Or I guess Docker Hub.

But really I think it's silly to push dev container images to a *public* registry when really we can push to the in-cluster registry...except what that runs straight up against is the self-signed router by default, which gets into https://github.com/openshift/machine-config-operator/pull/496 ... it'd really be nice though if we made it super convenient to use [Let's Encrypt](https://blog.openshift.com/requesting-and-installing-lets-encrypt-certificates-for-openshift-4/) or so.

File: docs/HACKING.md [hunk 1]
```diff
@@ -2,44 +2,16 @@
...
-These instructions have been tested inside a Fedora 29 podman container (on a FSB29 host).
+These instructions have been tested inside a Fedora 30 toolbox (podman) container (on a FSB30 host).
 It should also work to run these commands directly on a host system if you haven't yet
 containerized your workflow.  You will need build dependencies such as a `go` compiler,
 and the image builds rely on `podman`.
 
 ## Prerequisites
 
-1. Create a cluster using [the installer](https://github.com/openshift/installer/).  These instructions will be kept up to date generally against the leading edge of the installer.  Make sure you have set `KUBECONFIG` per the output of the installer.
+Most changes you'll want to test live against a real cluster.  Use [the installer](https://github.com/openshift/installer/), and in particular you'll want to use the "latest CI" release image.  As of the time of this writing, that's [4.2](https://openshift-release.svc.ci.openshift.org/).
```
Comment (kikisdeliveryservice) [inline docs/HACKING.md]: Use the "latest CI" release image of [the installer]..

File: docs/HACKING.md [hunk 2]
```diff
@@ -49,17 +21,9 @@ While you're still in the mode of testing builds, use `make`:
...
-When you want to push to the cluster, use the `deploy-` prefix, e.g.:
-
-```
-make deploy-daemon
-```
-
-(Like above, you can use `operator` or `controller` in place of `daemon`).
-
-Use `oc get pods -w` to watch for your new code to be deployed.
+See below for deploying changes to a live cluster.
```
Comment (kikisdeliveryservice) [inline docs/HACKING.md]: let's add link to the section here

File: hack/build-image [hunk 3]
```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/python3
```
Comment (kikisdeliveryservice) [inline hack/build-image]: python!! :smile:

**Comment by cgwalters** [comment]
Updated the doc per comments

**Comment by LorbusChris** [comment]
/approve
can you lgtm this, @kikisdeliveryservice ?

**Comment by cgwalters** [comment]
Using this some more, I now notice it's doing

```
--- /tmp/rendered-master-7922d771acbd9d212c63b49ce17ff0b4	2019-07-19 14:46:52.981967525 +0000
+++ /tmp/rendered-master-f3613c0c705e7cd9696b267237be420f	2019-07-19 14:47:10.607957263 +0000
@@ -2,10 +2,10 @@
 kind: MachineConfig
 metadata:
   annotations:
-    machineconfiguration.openshift.io/generated-by-controller-version: 55a73fe074fcec644f93a8abb1c96ea853fecb43
-  creationTimestamp: "2019-07-19T12:39:35Z"
+    machineconfiguration.openshift.io/generated-by-controller-version: 7be16b97ea8a5c1b9035bdea87b2e69561776087
+  creationTimestamp: "2019-07-19T14:40:30Z"
   generation: 1
-  name: rendered-master-7922d771acbd9d212c63b49ce17ff0b4
+  name: rendered-master-f3613c0c705e7cd9696b267237be420f
   ownerReferences:
   - apiVersion: machineconfiguration.openshift.io/v1
     blockOwnerDeletion: true
@@ -13,9 +13,9 @@
     kind: MachineConfigPool
     name: master
     uid: 36a50017-aa22-11e9-9e86-02eb8e87567e
-  resourceVersion: "3475"
-  selfLink: /apis/machineconfiguration.openshift.io/v1/machineconfigs/rendered-master-7922d771acbd9d212c63b49ce17ff0b4
-  uid: 43946e13-aa22-11e9-9e86-02eb8e87567e
+  resourceVersion: "43399"
+  selfLink: /apis/machineconfiguration.openshift.io/v1/machineconfigs/rendered-master-f3613c0c705e7cd9696b267237be420f
+  uid: 285dbd8f-aa33-11e9-be58-02615f8d79ce
 spec:
   config:
     ignition:
@@ -41,7 +41,7 @@
         mode: 420
         path: /etc/etcd/etcd.conf
       - contents:
-          source: data:,apiVersion%3A%20v1%0Akind%3A%20Pod%0Ametadata%3A%0A%20%20name%3A%20etcd-member%0A%20%20namespace%3A%20openshift-etcd%0A%20%20labels%3A%0A%20%20%20%20k8s-app%3A%20etcd%0Aspec%3A%0A%20%20initContainers%3A%0A%20%20-%20name%3A%20discovery%0A%20%20%20%20image%3A%20%22registry.svc.ci.openshift.orA
+          source: data:,apiVersion%3A%20v1%0Akind%3A%20Pod%0Ametadata%3A%0A%20%20name%3A%20etcd-member%0A%20%20namespace%3A%20openshift-etcd%0A%20%20labels%3A%0A%20%20%20%20k8s-app%3A%20etcd%0Aspec%3A%0A%20%20initContainers%3A%0A%20%20-%20name%3A%20discovery%0A%20%20%20%20image%3A%20%22image-registry.openshift-imaA
           verification: {}
         filesystem: root
         mode: 420
```

which triggers a master rollout.  Something to fix later.

**Pull request merged**